### DISCUSSION TITLE
Can't seem to get the fish_completer working, what am I missing?

### DISCUSSION THREAD

**Discussion by Ruudjhuu** [xonsh/xonsh#5004] [Q&A] [answered]
I tried to get the fish_completer working. I tried with xonsh installed by yay (on EndeavourOS) and by pip without any luck. If I understand correctly, fish_completer should be part of native xonsh so nothing additional should be installed (maybe fish). I also have fish installed. The output of xontrib list is as shown beneath but it still is not working. Which step am I missing here? How can I find out what is going wrong?

```
abbrevs             not-loaded
autovox             not-loaded
bashisms            not-loaded
coreutils           not-loaded
fish_completer      loaded      manual
free_cwd            not-loaded
pdb                 not-loaded
prompt_bar          not-loaded
prompt_starship     not-loaded
vox                 not-loaded
voxapi              not-loaded
whole_word_jumping  not-loaded
xog                 not-loaded
```

**Comment by jnoortheen** [comment]
Do you have fish installed in the path?

**Comment by Ruudjhuu** [comment, marked as answer]
Thanks for your answer, Yes fish is installed and the location is in the PATH

```
┬─[ruud@ruud-desktop:~]─[17:26:56]─[G:=]^M
╰─>$ whereis fish
<stdin>:1:8 - whereis fish
<stdin>:1:8 + ![whereis fish]
fish: /usr/bin/fish /etc/fish /usr/share/fish /usr/share/man/man1/fish.1.gz
┬─[ruud@ruud-desktop:~]─[17:27:05]─[G:=]^M
╰─>$ echo $PATH<stdin>:1:5 - echo $PATH
┬─[ruud@ruud-desktop:~]─[17:27:05]─[G:=]^M
╰─>$ echo $PATH
<stdin>:1:5 - echo $PATH
<stdin>:1:5 + ![echo $PATH]
/usr/local/sbin:/usr/local/bin:/usr/bin:/var/lib/flatpak/exports/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/ruud/.local/bin/:/home/ruud/.local/bin/

```

**Reply by jnoortheen** [reply]
you can use https://xon.sh/envvars.html#xonsh-trace-completions variable to see how the completions run.  Otherwise you can use `completer complete "git chec"` alias as well to see the generated completions

**Reply by Ruudjhuu** [reply]
Ah, I found it, the bash completer wasn't disabled an used before the fish completer. Thanks for the help. Adding `completer rm bash` in the xonshrc file fixed the issue.

**Reply by jnoortheen** [reply]
It is not necessary to remove it, if not explicitly added back. Loading this xontrib should place this above bash. 

https://github.com/xonsh/xonsh/blob/c07155dbe2a86c9547326e91baaea3c47511c370/xontrib/fish_completer.py#L30

You can check that by `completer list` and see whether fish is above bash
